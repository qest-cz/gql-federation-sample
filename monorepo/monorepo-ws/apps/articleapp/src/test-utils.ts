import { testServer } from '@monorepo-ws/server';
import { gql } from 'apollo-server';
import { modules } from './resolvers/modules';
import { GqlUser } from './resolvers/interfaces';
import { DataSources } from './main'; 
import { FakePrismaArticleDataSource } from './data-sources/fake-prisma-article-data-source';

const testModules = [
  {
    typeDefs: gql`
      extend type Query {
        getUser: User!
      }
    `,
    resolvers: {
      Query: {
        getUser: (): Pick<GqlUser, 'id'> => {
          return { id: 1 };
        },
      },
    },
  },
];

export const server = testServer([...modules, ...testModules], (): DataSources => ({article: new FakePrismaArticleDataSource()}));