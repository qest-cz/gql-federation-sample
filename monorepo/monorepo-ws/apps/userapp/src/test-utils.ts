import { testServer } from '@monorepo-ws/server';
import { modules } from './resolvers/modules';
import { DataSources } from './main'
import { FakePrismaUserDataSource } from './data-sources/fake-prisma-user-data-source';

export const server = testServer(modules, (): DataSources => ({user: new FakePrismaUserDataSource()}));