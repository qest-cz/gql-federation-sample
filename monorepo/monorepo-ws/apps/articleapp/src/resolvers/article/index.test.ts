import { gql } from "apollo-server";
import { server as testServer } from '../../test-utils'

const queryArticle = gql`
    query{
        articles{
            id,
            title,
            authorId
        }
    }
`
const getArticles = gql`
    query {
        getAllArticles {
            id,
            title,
            authorId
        }
    }
`

describe('article resolver', () => {
    it('should return articles', async () =>{
        const result = await testServer.executeOperation({
            query: queryArticle
        }) 
        expect(result.data).toMatchSnapshot({articles: [
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            },
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            },
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            }
        ]})        
    }),
    it('shoult return articles from getArticles', async () => {
        const result = await testServer.executeOperation({
            query: getArticles
        })
        expect(result.data).toMatchSnapshot({getAllArticles: 
            [
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            },
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            },
            {
                id: expect.any(String),
                title: expect.any(String),
                authorId: expect.any(String)
            }
        ]})
    })
})
