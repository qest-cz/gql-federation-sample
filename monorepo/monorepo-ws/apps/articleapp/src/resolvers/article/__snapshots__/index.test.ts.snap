// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article resolver should return articles 1`] = `
Object {
  "articles": Array [
    Object {
      "authorId": Any<String>,
      "id": Any<String>,
      "title": Any<String>,
    },
    Object {
      "authorId": Any<String>,
      "id": Any<String>,
      "title": Any<String>,
    },
  ],
}
`;

exports[`article resolver shoult return articles from getArticles 1`] = `
Object {
  "getAllArticles": Array [
    Object {
      "authorId": Any<String>,
      "id": Any<String>,
      "title": Any<String>,
    },
    Object {
      "authorId": Any<String>,
      "id": Any<String>,
      "title": Any<String>,
    },
  ],
}
`;
